### PR TITLE
Set up github actions configuration

### PR DESCRIPTION
**Pull request by baldwinjj**
@mdzeng 

Set up github actions configuration

cc @vincecampanale

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,19 +0,0 @@
...
-branches:
-  only:
-    - master
-    - "/^v?([0-9]+)\\.([0-9]+)\\.([0-9]+)$/"
-    - '/^feature.*/'
-    - '/^chore.*/'
-    - '/^bug.*/'
-    - '/^revert.*/'
-script:
-  - npm run test:travis
```
Comment (mdzeng) [inline .travis.yml:14]: Can we remove this from package.json as well?

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +0,0 @@
-language: node_js
-node_js:
-  - 16.13.0
-cache: yarn
-branches:
-  only:
```
Comment (mdzeng) [inline .travis.yml:6]: This logic doesn't look like it's replaced in the github action workflow?

**Review by mdzeng — COMMENTED** [review]
Looks good, just had a couple of questions!

File: .travis.yml (hunk 2 above)
Comment (baldwinjj) [reply .travis.yml:6]: This seemed like legacy configuration. I could be wrong, but lint/test is cheap so why not run it on all branches?

Comment (nanek) [reply .travis.yml:6]: I've been removing it. I'm not sure it provides value anymore. We still name branches that way, but don't need to restrict github actions.

**Review by mdzeng — APPROVED** [review]

File: .travis.yml (hunk 2 above)
Comment (sandinmyjoints) [reply .travis.yml:6]: There are some tools that hang a regex off these branch name conventions to extract ticket numbers from the branch names. A pre-commit git hook and changebot (the Slack bot that posts to #deployments) come to mind. I slightly favor keeping the enforcement via CI to steer folks to the patterns so the tooling continues to work.

Comment (baldwinjj) [reply .travis.yml:6]: The neodarwin configuration is more restrictive by actually failing the build if the branch isn't named with the expected pattern. The travis configuration here was just bypassing CI altogether if the branch was named something other than the expected branch names. Do we want more restrictive enforcement?

Comment (nanek) [reply .travis.yml:6]: Referenced scripts https://github.com/spanishdict/playbook/blob/main/development-setup/git-workflow.md#git-hooks
https://github.com/spanishdict/neodarwin/blob/main/bin/changelog-generate

I believe these just look for 8 or 9 digits and then assume that is the PT id. The branch prefixes are no longer considered.

Comment (sandinmyjoints) [reply .travis.yml:6]: Oh, nice! Thanks @nanek , I didn't realize that. I need to update my own git hooks locally. Anyway, I still like the branch conventions but no objections to removing enforcement.

**Pull request merged**